plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/RemoteTableName.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.bigquery;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.cloud.bigquery.TableId;
+
+import java.util.Objects;
+
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public final class RemoteTableName
```

Comment:
How is that different than TableId?
Maybe we should have single class but add a boolean `resolved` flag to mark if the names inside match exactly the case used by big query?

Reply:
I can make this class extend TableId but I don't like the BigQuery models leaking out of BigQueryClient.

My first cut was similar (I used the existing `project`, `dataset` and `table` fields on the TableHandle) but the issue was that you need to remember which methods need the lowercase values and which ones need the remote values.

The boolean flag will only save calling `toRemoteXXX` methods and save some remote API calls. Or did you mean something else?

Reply:
Never mind. I think I was not aware that `TableId` was coming from bigquery client library. It is ok.